feat: new tailwind preset package (#3829)
kiwicom/orbit: packages/orbit-design-tokens/docs/SCHEMA.md

Therefore, we use a structured and standardized way to create, store and work with Orbit design tokens.

## Namespace / Object / Variant / Property

In similarity with the **Category / Type / Item** schema, defined by [style-dictionary](https://amzn.github.io/style-dictionary/#/), we aim to have design tokens defined by some hierarchical tree structure.

This hierarchical tree structure brings a consistent approach to how our design tokens are being defined in a systematic way. Therefore, you will see that every definition of our design tokens will look like this:

```json
{
  "component": {
    "button": {
      "large": {
        "height": {
          "type": "size",
          "value": "{foundation.size.extra-large}"
        }
      }
    }
  }
}
```

The above example is being translated with custom attribute transforms. For information on how we work with the `namespace.object.variant.property` hierarchical tree structure, check the transforms document.

### Namespace

In our design tokens, a `namespace` is considered the top-level layer of a token. It determines the intention of the token – where and how exactly it will be used.

> Possible values of the `namespace":

Reviewer: ```suggestion
Possible values of the `namespace`:
```

# Schema of @kiwicom/orbit-design-tokens

Orbit design tokens could be considered as a simple way to describe, document and use the visual language of Orbit design system, but there are multiple requirements that it needs to meet and support.

> Therefore, we use structured and standardized way Orbit design tokens are being created, stored and how we work with them.

Reviewer: I am not sure what this intends to say 😕 Maybe this?

```suggestion
Therefore, we use a structured and standardized way to create, store and work with Orbit design tokens.
```

## Namespace / Object / Variant / Property

In similarity with **Category / Type / Item** defined by style-dictionary we aim to have design tokens defined by some hierarchical tree structure.

This hierarchical tree structure brings consistent approach how our design tokens are being defined in systematical way. Therefore, you will see that every definition of our design token will look like this:

```json
{
  "component": {
    "button": {
      "large": {
        "height": {
          "type": "size",
          "value": "{foundation.size.extra-large}"
        }
      }
    }
  }
}
```

The above example is being translated with custom attribute transforms. For information how we work with `namespace.object.variant.property` hierarchical tree structure check transforms document.

### Namespace

In our design tokens an `namespace` is considered the top-level layer of design tokens that determines the intention of the token – where and how exactly it will be used.

> Possible values of the object level are:

Author: Yep

- The `foundation` is being used for the internal design tokens that are used as aliases for `global` and `component` design tokens, and/or interpretation of values that can be overwritten by custom theming,
- the `global` is being used for the most common design tokens that interprets the fundamental visual language of the Orbit design system,
- the `component` is being used for grouped design tokens that can be used for similar components, e.g. shared design token for the height of the buttons and/or inputs.

### Object

> The `object` is an attribute to specify either the name of some component (component group) or [foundation object](#foundation-objects).

Author: Object here is not an object in classing meaning, above it's described the structure like: 

`## Namespace / Object / Variant / Property`

so for instance, in case of the size foundation: 

```
   "foundation": {
    "size": {
      "small": {
        "type": "size",
        "value": 16,
        "internal": true
      },
    .... 
```

where the foundation is the namespace and "size" is the Object that contains normal object

> The `object` is an attribute to specify either the name of some component (component group) or [foundation object](#foundation-objects).

Author: feel free to suggest the alternative :) I also just moved that part from the old MR

> The `object` is an attribute to specify either the name of some component (component group) or [foundation object](#foundation-objects).

Reviewer: Ahh, understood. I got confused by `object` (the JS primitive) and the name `Object` (used in this tokens scope) 😅 I think it is acceptable as is, but I would personally prefer something like

```
The `object` attribute is used to specify either the name of some component (component group) or the [foundation object](#foundation-objects).
```

Just because I think it leaves no doubt between the two _objects_. But I will leave the decision for you.

### Property (or either sub-variant)

Following after the `variant` attribute, there is a need to specify either `property` (e.g. `height`) or sometimes allowed sub-variants for some objects. This is especially handy when the token is more complex and has more specificity, like [additional palette shades](#list-of-values-for-variant-and-sub-variant-attributes-of-palette-shades).

## Design token definition types

Because design tokens are data representation of the visual language, we need to use many types of them. All the following types of design tokens have their transformers available either by the default implementation from `style-dictionary` or by custom implementation. For more information about the architecture check the [...]() document.

- border-radius
- color
- breakpoint
- duration
- box-shadow
- size
- font-family
- font-weight
- z-index
- modifier
- spacing
- opacity

## Additional attributes

> For some cases we define and use additional attributes that help us maintain the whole structure of Orbit design tokens.

Author: I will remove that paragraph, because currently it's just useless

To be able to determinate if `normal` should be used, following conditions should be used:

- use `medium`, when there is no preferred size for the variant or attribute,
- use `normal`, when it should be the default variant/size and used in the majority of possible cases.

### Variant types

Orbit design tokens use many contextual types – from statuses (e.g. success) to simple importance variant (e.g. primary).

Status types:
– info
– success

- warning
  – critical
- neutral

Ordered importance types:

- primary
  – secondary
  – tertiary

We also use additional types connected to some specific behaviour, such as:

- selected
- compact

And lastly, there can be types connected to visual language:

- inverted
- white
- dark
- subtle

### Duration of transitioning

For transitioning, we use simple scale of three values that represents relative speed:

– slow
– normal

- fast

### Typography scale

For typography, we use combination of the [sizing scale](#sizing-scale) and scale that represents the visual hierarchy of heading levels:

- display
- displaySubtitle
- title1 to title5 according to existing HTML heading tags

### Elevation scale

> Based on our [elevation system](https://orbit.kiwi/foundation/elevation-hierarchy/), we use only following naming variants for the definition of them:

Author: elevations :)

- suppressed
- flat
- action
- fixed
- raised
- overlay

### Viewports

> smallMobile, mediumMobile, largeMobile, tablet, desktop, largeDesktop

Author: Do not think so, our docs are already covering that
